src/context.rs

```diff
@@ -39,10 +39,18 @@ pub fn get_buffer_words(buf: &Buffer) -> Vec<(usize, usize)> {
     res
 }
 
+/// The key bindings to use.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum KeyBindings {
+    Vi,
+    Emacs,
+}
+
 pub struct Context {
     pub history: History,
     pub completer: Option<Box<Completer>>,
     pub word_fn: Box<Fn(&Buffer) -> Vec<(usize, usize)>>,
+    pub key_bindings: KeyBindings,
```

Comment:
setting this might work better behind a method. Right now we create a new instance of `KeyMap` for each call to `read_line()` but if the `KeyMap` wanted to maintain state between invocations of `read_line()` (say, to implement copy and paste for example), that won't work. Could implement an `enum` to contain all of the keymaps and then change it via a method that accepts `KeyBindings` as an argument. I need to think about it. Feel free to add your input as well.